src/wolff_producer.erl

```diff
@@ -116,40 +129,50 @@ stop(Pid) ->
 -spec send(pid(), [wolff:msg()], wolff:ack_fun()) -> ok.
 send(Pid, [_ | _] = Batch0, AckFun) ->
   Caller = self(),
-  Mref = erlang:monitor(process, Pid),
```

Comment:
The monitor/2 is a very heavy operation on a machine with many CPU cores. I have tested on a 64-core ubuntu, with this monitoring it consumes about twice as more CPU as the non-monitor one.

Reply:
how is non-mornitor implemented ?

Reply:
Finally I figured out that the `monitor/2` is as expensive as sending a message to the producer. In my testing, when ETS batcher is disabled, simply removing the `monitor/2` and `demonitor/2` calls improves almost nothing as we have to call the `erlang:send/2` anyway.

When ETS batcher enabled, it is not necessary to monitoring the process (and no message passing is needed), so the performance is significantly improved comparing to the no ETS batcher one.

With the ETS batcher CPU usage is decreased by 30%. The Erlang VM schedulers was bound to the same physical CPU core which has 8 logical threads. If the schedulers unbound, the usage can be reduced to 50% of the one without ETS batcher.